src/utilities/i18n/tests/I18n.test.ts

```diff
@@ -17,12 +17,17 @@ describe('I18n.translate', () => {
   });
 
   it('merges multiple dictionaries', () => {
-    const i18n = new I18n([
-      {one: 'one', two: 'two', three: 'three', fallback: 'fallback'},
+    const translations: Record<string, string>[] = [
       {one: 'uno', two: 'deux', three: 'trois'},
-    ]);
+      {one: 'one', two: 'two', three: 'three', fallback: 'fallback'},
+    ];
+
+    const i18n = new I18n(translations);
     expect(i18n.translate('one')).toBe('uno');
     expect(i18n.translate('fallback')).toBe('fallback');
+
+    // Assert the original translations array is not mutated
+    expect(translations[0].two).toBe('deux');
```

Comment:
Should this be in a separate test?

Reply:
I don't think it's worth it. I don't see much value in having multiple tests with exactly the same setup but slightly different assertions, I'd rather stick them into one test - that stops you wasting time repeating the setup